flink-clients/src/main/java/org/apache/flink/client/deployment/application/executors/EmbeddedExecutor.java

```diff
@@ -153,7 +173,14 @@ private CompletableFuture<JobClient> submitAndGetJobClientFuture(
                                     return jobId;
                                 }))
                 .thenApplyAsync(
-                        jobID -> jobClientCreator.getJobClient(actualJobId, userCodeClassloader));
+                        jobID -> jobClientCreator.getJobClient(actualJobId, userCodeClassloader))
+                .whenCompleteAsync(
+                        (jobClient, throwable) -> {
+                            if (throwable == null) {
```

Comment:
Discussed with @davidradl offline. The throwable is not able to rethrow in whenCompleteAsync. Thus, log the exception rather than swallow the throwable.
https://stackoverflow.com/questions/71668871/completablefuture-whencompleteasync-does-not-let-me-re-throw-an-exception